File: 2015/Day10/main.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

string look_and_say(const string &input)
{
    string result;
    int count = 1;

    for (size_t i = 1; i <= input.length(); ++i)
    {
        if (i < input.length() && input[i] == input[i - 1])
        {
            count++;
        }
        else
        {
            result += to_string(count) + input[i - 1];
            count = 1;
        }
    }

    return result;
}

int main()
{
    string input = "3113322113";

    for (int i = 0; i < 40; ++i)
    {
        input = look_and_say(input);
    }

    cout << "Part 1: " << input.length() << endl;

    for (int i = 0; i < 10; ++i)
    {
        input = look_and_say(input);
    }

    cout << "Part 2: " << input.length() << endl;

    return 0;
}

File: 2015/Day10/main.py
def look_and_say(input_str):
    result = []
    count = 1
    for i in range(1, len(input_str) + 1):
        if i < len(input_str) and input_str[i] == input_str[i - 1]:
            count += 1
        else:
            result.append(str(count))
            result.append(input_str[i - 1])
            count = 1
    return "".join(result)

input_str = "3113322113"

for _ in range(40):
    input_str = look_and_say(input_str)

print(f"Part 1: {len(input_str)}")

for _ in range(10):
    input_str = look_and_say(input_str)

print(f"Part 2: {len(input_str)}")
